Assets/Scripts/Managers/TaskManager.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections;
+using UnityEngine;
+
+namespace Managers
+{
+	internal class TaskManager : MonoBehaviour
```

Comment:
You could use the Singleton class instead
```suggestion
	internal class TaskManager : Singleton<TaskManager>
```

Reply:
I hadnt seen that the singleton could create itself, then the only thing that is stopping me from immediately switching to that is a little bug that singleton has whereby its not a singleton
![image](https://github.com/Taliayaya/Project-86/assets/85966398/2225202b-a052-4d1a-8ed7-e6170d15ad69)

Reply:
What kind of bug is it, is it multiplying itself? (if yes, I haven't seen it happen and still can't) Or the fact that multiple singleton are stacked on the same go?

Reply:
When reloading the scene, it just creates a new one because it is a physical scene object and it doesn't check on awake if there is already one so if

created one (entered playmode)
access it (so that _instance is set)
create as many unregistered ones as you want

thats why i usually go for the [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)] method, but a check in awake also works, just with a little extra overhead

I can add the check in awake if you want cause i need to move it out of DefaultNamespace anyways cause it was causing issues with commits [786a3b7](https://github.com/Taliayaya/Project-86/pull/39/commits/786a3b7c847c8e84689ab8ca98aaba183379f8c8), [20671d7](https://github.com/Taliayaya/Project-86/pull/39/commits/20671d767be18ff7ac152cae1f1e217b66fa9f26), and
 [8aae355](https://github.com/Taliayaya/Project-86/pull/39/commits/8aae3554d3ff758a682ef3478d4d56f0ed8518ee)

Reply:
Oh that is probably explaining some bugs then. I never paid attention to be honest. You can modify it. If you want you can also remove the OnApplicationQuit that returns null instead of the instance. 
It makes more sense now.